operations/helm/charts/alloy/templates/containers/_agent.yaml

```diff
@@ -57,6 +57,10 @@
       scheme: {{ $values.listenScheme }}
     initialDelaySeconds: 10
     timeoutSeconds: 1
+  {{- with $values.lifecycle }}
+  lifecycle:
```

Comment:
Is this supposed to set a `lifecycle` on the Pod spec?

Reply:
```
% kubectl explain pod.spec.lifecycle
KIND:       Pod
VERSION:    v1

error: field "lifecycle" does not exist
```

Reply:
➜  ~ `kubectl explain pod.spec.containers.lifecycle`
KIND:       Pod
VERSION:    v1

FIELD: lifecycle <Lifecycle>


DESCRIPTION:
    Actions that the management system should take in response to container
    lifecycle events. Cannot be updated.
    Lifecycle describes actions that the management system should take in
    response to container lifecycle events. For the PostStart and PreStop
    lifecycle handlers, management of the container blocks until the action is
    complete, unless the container process fails, in which case the handler is
    aborted.

FIELDS:
  postStart	<LifecycleHandler>
    PostStart is called immediately after a container is created. If the handler
    fails, the container is terminated and restarted according to its restart
    policy. Other management of the container blocks until the hook completes.
    More info:
    https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#container-hooks

  preStop	<LifecycleHandler>
    PreStop is called immediately before a container is terminated due to an API
    request or management event such as liveness/startup probe failure,
    preemption, resource contention, etc. The handler is not called if the
    container crashes or exits. The Pod's termination grace period countdown
    begins before the PreStop hook is executed. Regardless of the outcome of the
    handler, the container will eventually terminate within the Pod's
    termination grace period (unless delayed by finalizers). Other management of
    the container blocks until the hook completes or until the termination grace
    period is reached. More info:
    https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#container-hooks